sql/parser/sql.y

```diff
@@ -1003,13 +1005,19 @@ set_reset_clause:
   SET set_rest {}
 
 show_stmt:
-  SHOW var_name
+  SHOW var_name on_privilege_target_clause for_grantee_clause
```

Comment:
I also just realized that my hacky "if $3 != nil" should be an error, not nil. I can't find an easy way to surface an error from the grammar, so I guess I'll need to do this properly.